base/compiler/tfuncs.jl

```diff
@@ -1699,6 +1709,87 @@ function _builtin_nothrow(@nospecialize(f), argtypes::Array{Any,1}, @nospecializ
     return false
 end
 
+# known to be always effect-free (in particular nothrow)
+const _PURE_BUILTINS = Any[tuple, svec, ===, typeof, nfields]
```

Comment:
I wonder if it would make more sense to declare these const as SimpleVectors:
```suggestion
const _PURE_BUILTINS = svec(tuple, svec, ===, typeof, nfields)
```